src/uncrustify.cpp

```diff
@@ -1416,8 +1418,8 @@ static void add_func_header(c_token_t type, file_mem &fm)
 
       if (ref->type == CT_FUNC_DEF && ref->parent_type == CT_NONE && ref->next)
       {
-         int found_brace = 0; // Set if a close brace is found before a newline
-         while (ref->type != CT_NEWLINE && (ref = ref->next))
+         int found_brace = 0;                                   // Set if a close brace is found before a newline
+         while ((ref->type != CT_NEWLINE) && (ref = ref->next)) /* \todo is the assignment of ref wanted here?, better move it to the loop */
```

Comment:
Yeah, I agree with @todd-richmond here. On the other hand, when the expression is not clear and it requires knowledge about C++ operator precedence then I would add parentheses.